### ISSUE TITLE
Markup Support

### ISSUE THREAD

**Issue by dbrgn** [blizzard4591/openMittsu#79]
As mentioned in #9: Threema has switched away from pure regexes towards a proper parser.

The JavaScript version is published here: https://github.com/threema-ch/threema-markup Maybe that could be adapted, it's relatively simple and contains a big test suite.

**Comment by blizzard4591** [comment]
Yes, I looked at that already and it would make sense to adapt. My plan would be to transpile it to C++, as a manual adaption makes no sense in my eyes, tracking changes just does not scale or make sense manually.
Sadly, the TypeScript to C++ transpiler I found can not handle the used features: https://github.com/ASDAlexander77/TypeScript2Cxx/issues/7

I am open to other ideas, but the generated workload needs to be in-line with this being a one-person show.
